src/update_agent/actor.rs

```diff
@@ -402,28 +385,20 @@ impl UpdateAgentInfo {
                 ));
                 // Record postponement and postpone finalization.
                 state.record_postponement();
-                Err(())
-            } else {
-                self.finalize_deployment(release).await
+            } else if let Ok(release) = self.finalize_deployment(release).await {
```

Comment:
This one is a strange mix where half of the result handing is done in the function itself (error logging) and half is done here at the caller (update on success).
It may be worth trying to move this one too to plain `async` too, and then decide whether to do the whole `Result` handling either fully in the function or fully here.

Reply:
I just moved `Result` handling fully here, but noticed that this made the `self.finalize_deployment` function slightly different from the other similar helpers. In the other helpers, such as `attempt_deploy`,  it does the error logging in the function itself, but the caller _also_ does some additional handling.
For the case of `finalize_deployment` here, there is nothing additional to do for the caller if the result is `Err`. So would it actually make more sense to leave this as is, or perhaps we should move the error logging to the caller function in the other helpers as well?

Reply:
On second thought, yeah I think it would make sense to just move all the error logging to the callers.